superset/__init__.py

```diff
@@ -14,231 +14,37 @@
-
-        def __call__(self, environ, start_response):
-            # Setting wsgi.input_terminated tells werkzeug.wsgi to ignore
-            # content-length and read the stream till the end.
-            if environ.get("HTTP_TRANSFER_ENCODING", "").lower() == u"chunked":
-                environ["wsgi.input_terminated"] = True
-            return self.app(environ, start_response)
-
-    app.wsgi_app = ChunkedEncodingFix(app.wsgi_app)  # type: ignore
-
-if app.config["UPLOAD_FOLDER"]:
-    try:
-        os.makedirs(app.config["UPLOAD_FOLDER"])
-    except OSError:
-        pass
-
-for middleware in app.config["ADDITIONAL_MIDDLEWARE"]:
-    app.wsgi_app = middleware(app.wsgi_app)  # type: ignore
-
-
-class MyIndexView(IndexView):
-    @expose("/")
-    def index(self):
-        return redirect("/superset/welcome")
-
-
-custom_sm = app.config.get("CUSTOM_SECURITY_MANAGER") or SupersetSecurityManager
-if not issubclass(custom_sm, SupersetSecurityManager):
-    raise Exception(
-        """Your CUSTOM_SECURITY_MANAGER must now extend SupersetSecurityManager,
-         not FAB's security manager.
-         See [4565] in UPDATING.md"""
-    )
-
-with app.app_context():
-    appbuilder = AppBuilder(
-        app,
-        db.session,
-        base_template="superset/base.html",
-        indexview=MyIndexView,
-        security_manager_class=custom_sm,
-        update_perms=False,  # Run `superset init` to update FAB's perms
-    )
-
-security_manager = appbuilder.sm
-
-results_backend = app.config.get("RESULTS_BACKEND")
-results_backend_use_msgpack = app.config.get("RESULTS_BACKEND_USE_MSGPACK")
-
-# Merge user defined feature flags with default feature flags
-_feature_flags = app.config.get("DEFAULT_FEATURE_FLAGS") or {}
-_feature_flags.update(app.config.get("FEATURE_FLAGS") or {})
-
-# Event Logger
-event_logger = get_event_logger_from_cfg_value(
-    app.config.get("EVENT_LOGGER", DBEventLogger())
-)
-
-
-def get_feature_flags():
-    GET_FEATURE_FLAGS_FUNC = app.config.get("GET_FEATURE_FLAGS_FUNC")
-    if GET_FEATURE_FLAGS_FUNC:
-        return GET_FEATURE_FLAGS_FUNC(deepcopy(_feature_flags))
-    return _feature_flags
-
-
-def is_feature_enabled(feature):
-    """Utility function for checking whether a feature is turned on"""
-    return get_feature_flags().get(feature)
-
-
-# Flask-Compress
-if conf.get("ENABLE_FLASK_COMPRESS"):
-    Compress(app)
-
-
-talisman = Talisman()
-
-if app.config["TALISMAN_ENABLED"]:
-    talisman.init_app(app, **app.config["TALISMAN_CONFIG"])
-
-# Hook that provides administrators a handle on the Flask APP
-# after initialization
-flask_app_mutator = app.config.get("FLASK_APP_MUTATOR")
-if flask_app_mutator:
-    flask_app_mutator(app)
-
-from superset import views  # noqa isort:skip
-
-# Registering sources
-module_datasource_map = app.config["DEFAULT_MODULE_DS_MAP"]
-module_datasource_map.update(app.config["ADDITIONAL_MODULE_DS_MAP"])
-ConnectorRegistry.register_sources(module_datasource_map)
+app = create_app()
```

Comment:
We do and we don't. My goal is to clear it out completely... However, the changes required in order to facilitate this will end up being quite a large PR. If you think it's reasonable to "just do it", then I'd be happy to pull that thread.

One thing of note here is that removing `app` from here will REQUIRE people to register their `FLASK_APP` env var differently `superset.app` -> `superset.app:create_app()`